internal/attestation/measurements/measurements.go

```diff
@@ -94,58 +70,224 @@ func (m *M) FetchAndVerify(ctx context.Context, client *http.Client, measurement
 
 // CopyFrom copies over all values from other. Overwriting existing values,
 // but keeping not specified values untouched.
-func (m M) CopyFrom(other M) {
+func (m *M) CopyFrom(other M) {
 	for idx := range other {
-		m[idx] = other[idx]
+		(*m)[idx] = other[idx]
 	}
 }
 
 // EqualTo tests whether the provided other Measurements are equal to these
 // measurements.
-func (m M) EqualTo(other M) bool {
-	if len(m) != len(other) {
+func (m *M) EqualTo(other M) bool {
+	if len(*m) != len(other) {
 		return false
 	}
-	for k, v := range m {
-		if !bytes.Equal(v, other[k]) {
+	for k, v := range *m {
+		otherExpected := other[k].Expected
+		if !bytes.Equal(v.Expected[:], otherExpected[:]) {
+			return false
+		}
+		if v.WarnOnly != other[k].WarnOnly {
 			return false
 		}
 	}
 	return true
 }
 
-// MarshalYAML overwrites the default behaviour of writing out []byte not as
-// single bytes, but as a single base64 encoded string.
-func (m M) MarshalYAML() (any, error) {
-	base64Map := make(map[uint32]string)
+// GetEnforced returns a list of all enforced Measurements,
+// i.e. all Measurements that are not marked as WarnOnly.
+func (m *M) GetEnforced() []uint32 {
+	var enforced []uint32
+	for idx, measurement := range *m {
+		if !measurement.WarnOnly {
+			enforced = append(enforced, idx)
+		}
+	}
+	return enforced
+}
+
+// SetEnforced sets the WarnOnly flag to true for all Measurements
+// that are NOT included in the provided list of enforced measurements.
+func (m *M) SetEnforced(enforced []uint32) error {
+	enforcedMap := map[uint32]struct{}{}
+	for _, idx := range enforced {
+		enforcedMap[idx] = struct{}{}
+	}
+
+	// set all measurements to warn only
+	for idx, measurement := range *m {
+		(*m)[idx] = Measurement{
+			Expected: measurement.Expected,
+			WarnOnly: true,
+		}
+	}
 
-	for key, value := range m {
-		base64Map[key] = base64.StdEncoding.EncodeToString(value[:])
+	// set enforced measurements from list
+	for _, idx := range enforced {
+		measurement, ok := (*m)[idx]
+		if !ok {
+			return fmt.Errorf("measurement %d not in list, but set to enforced", idx)
```

Comment:
Suggest rearranging the code to do the check first so that m is not changed on error. This may not be needed for the current users of M, but avoids surprises and is a good practice in general.